### PR TITLE
feat: add translation support

### LINKED ISSUE #789: Translation
**- Do you want to request a *feature* or report a *bug*?**
feature

**- What is the current behavior?**
the cms is only in english
**- If the current behavior is a bug, please provide the steps to reproduce.**

**- What is the expected behavior?**
it might be possible to select the default language in Netlify and also be able to change it in CMS.
**- Please mention your CMS, node.js, and operating system version.**

### PR DESCRIPTION
**Pull request by erezrokah**
Fixes #789 

Inspired by https://github.com/netlify/netlify-cms/pull/2528 and https://github.com/netlify/netlify-cms/pull/2278.

Similar mechanism as https://github.com/date-fns/date-fns/blob/aa1daea6a8a1493429330cdd82ae7c325a97b902/src/index.js

@erquhart I added a new package for the locales, I'm not sure how to go about versioning and publishing it

### CODE REVIEW

**Comment by barthc** [comment]
I assume the new package `netlify-cms-locales` would be bundled together with the app, no ?. Could increase bundle size as we add more locales. 
Based on this comment https://github.com/netlify/netlify-cms/issues/789#issuecomment-522767000 I am thinking we are looking for something similar to the way we load external config via url.

**Comment by erezrokah** [comment]
Hey @barthc, not necessarily in case you're using WebPack with the `netlify-cms-app` non registered locales will be tree shaked similar to the external media libraries: https://www.netlifycms.org/blog/2019/07/netlify-cms-gatsby-plugin-4-0-0/#using-media-libraries-with-netlify-cms-app

**Comment by chrfritsch** [comment]
I like the approach here. I also like that all supported languages are managed inside this repo. That makes contributing new translations quite easy.

I will open a PR later today to provide the german translations.

**Comment by chrfritsch** [comment]
During working on the german translation I realized that translating widget strings is currently not possible. For example "Choose an image"

**Comment by chrfritsch** [comment]
Here is the PR with the german translation https://github.com/netlify/netlify-cms/pull/2877

**Comment by erezrokah** [comment]
> During working on the german translation I realized that translating widget strings is currently not possible. For example "Choose an image"

Can you please open a separate issue for it?
Looks like this is the relevant place in the code to fix it: https://github.com/netlify/netlify-cms/blob/4833f337280dee0105b282c4863b3b674c818cba/packages/netlify-cms-widget-file/src/withFileControl.js#L249

File: packages/netlify-cms-app/package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "netlify-cms-widget-select": "^2.4.3",
     "netlify-cms-widget-string": "^2.2.3",
     "netlify-cms-widget-text": "^2.2.3",
+    "netlify-cms-locales": "^0.0.1",
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply packages/netlify-cms-app/package.json]: I'll change it to `1.0.0`

**Comment by erezrokah** [comment]
Updated the package version and registered all locales for `netlify-cms`.
The english translation size is 6.4K (2K gzipped) so I wouldn't worry about bundle size atm.
Also this https://github.com/netlify/netlify-cms/pull/2875 just shaved a 100K (unzipped) off the bundle.

**Pull request merged**